### PR TITLE
05 get /api/articles/:article_id

### PR DESCRIPTION
**Pull request by siandaniel**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -12,6 +12,19 @@ afterAll(() => {
     return db.end();
 });
 
+describe("/non-existent-or-misspelt-endpoint", () => {
+    describe("ALL REQUESTS", () => {
+        test("Returns 'Status: 404' and custom error message if endpoint is not found", () => {
+            return request(app)
+            .get('/invalid-or-misspelt-path')
+            .expect(404)
```
Comment (Rosem10) [inline __tests__/app.test.js:20]: Lovely!

File: __tests__/app.test.js [hunk 2]
```diff
@@ -125,14 +129,73 @@ describe("/api/articles", () => {
...
+});
+
+describe("/api/articles/:article_id", () => {
+    describe("GET", () => {
+        test("Returns 'Status: 200' with single article object if valid article ID", () => {
+            return request(app).get('/api/articles/1').expect(200)
+            .then(({ body }) => {
+                const article = body.requestedArticle;
+                expect(typeof article).toBe("object");
+                expect(Array.isArray(article)).toBe(false);
```
Comment (Rosem10) [inline __tests__/app.test.js:142]: Nice!

File: __tests__/app.test.js [hunk 3]
```diff
@@ -125,14 +129,73 @@ describe("/api/articles", () => {
...
+                expect(article).toHaveProperty("title", expect.any(String));
+                expect(article).toHaveProperty("article_id", expect.any(Number));
+                expect(article).toHaveProperty("body", expect.any(String));
+                expect(article).toHaveProperty("topic", expect.any(String));
+                expect(article).toHaveProperty("created_at", expect.any(String));
+                expect(article).toHaveProperty("votes", expect.any(Number));
+                expect(article).toHaveProperty("article_img_url", expect.any(String));
+            });
+        });
+        test("Returns the correct article object for article ID 1", () => {
```
Comment (Rosem10) [inline __tests__/app.test.js:159]: I would argue that you don't need this test as well as the top one, OR that you could just do expect article to equal (and then just the entire object)

File: __tests__/app.test.js [hunk 4]
```diff
@@ -125,14 +129,73 @@ describe("/api/articles", () => {
...
+                expect(article).toHaveProperty("created_at", "2020-07-09T20:11:00.000Z");
+                expect(article).toHaveProperty("votes", 100);
+                expect(article).toHaveProperty("article_img_url", "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700");
+            });
         });
-        test("Returns custom error message if invalid path provided ", () => {
-            return request(app).get('/api/articlees').expect(404).then((body) => {
-                const errorMsg = body.error.text
-                expect(errorMsg).toBe("Invalid path provided - please try again");
+        test("Returns the correct article object for other valid article IDs", () => {
```
Comment (Rosem10) [inline __tests__/app.test.js:175]: There's no need to do multiple tests for this - as long as it works for one, you've proven the concept!

File: __tests__/app.test.js [hunk 5]
```diff
@@ -125,14 +129,73 @@ describe("/api/articles", () => {
...
+        test("Returns 'Status: 400' and relevant error message if article ID is of incorrect data type", () => {
+            return request(app).get('/api/articles/abc').expect(400)
+            .then(({ body }) => {
+                expect(body.msg).toBe("Bad request - invalid data type for article ID");
+            });
+        });
+        test("Returns 'Status: 404' and relevant error message if article ID does not exist in database", () => {
+            return request(app).get('/api/articles/392').expect(404)
+            .then(({ body }) => {
+                console.log(body)
```
Comment (Rosem10) [inline __tests__/app.test.js]: Remove this console.log please!

File: app/controllers.js [hunk 6]
```diff
@@ -7,9 +7,19 @@ const getTopics = (request, response, next) => {
 };
 
 const getArticles = (request, response, next) => {
-        fetchArticles().then((articles) => {
-            response.status(200).send({ articles: articles });       
-        });
+    fetchArticles().then((articles) => {
+        response.status(200).send({ articles: articles });
```
Comment (Rosem10) [inline app/controllers.js:11]: You could shorten this to just {articles} if you wanted (it's shorthand for the same thing!)

File: app/models.js [hunk 7]
```diff
@@ -19,4 +19,18 @@ const fetchArticles = () => {
     });
 };
 
-module.exports = { fetchTopics, fetchArticles };
+const fetchArticleById = (article_id) => {
+    let sqlFetchArticleByIdQuery = `SELECT * FROM articles
+                                    WHERE article_id = $1`
+
+    return db.query(sqlFetchArticleByIdQuery, [article_id]).then(({ rows, rowCount }) => {
```
Comment (Rosem10) [inline app/models.js:26]: Absolutely LOVELY use of rowCount

**Review by Rosem10 — APPROVED** [review]
Really nice stuff, Sian! There are a couple of points to look at here before merging, but no need to submit another PR for it!

File: __tests__/app.test.js (hunk 3 above)
Comment (siandaniel) [reply __tests__/app.test.js:159]: Thank you, done the latter :)
